src/components/UpcomingEvents/index.jsx

```diff
@@ -145,27 +144,26 @@ const UpcomingEventsSection = () => {
     );
   }
 
-  if (fetchingDataError) {
-    return <div>{fetchingDataError} : an Error Occured While Loading</div>;
-  }
-
-  return (
-    <section className="upcoming-events-section">
-      <div className="upcoming-events-section__container lg:grid gap-8 grid-cols-12 container">
-        <div className="mb-8 col-start-1 col-end-8 pr-8" ref={ref}>
-          <UpcomingEvents data={data} slideTop={slideTop} />
+  if (data) {
```

Comment:
Please destructure the keys you are going to use to render the component here for readability reasons (you may refer to magdy or nag3awy for more details)